cpp/src/structs/utilities.cu

```diff
@@ -56,6 +58,103 @@ std::vector<std::vector<column_view>> extract_ordered_struct_children(
   return result;
 }
 
+/**
+ * @brief Flattens struct columns to constituent non-struct columns in the input table.
+ *
+ */
+struct flatten_table {
```

Comment:
Seems like as this is a `struct` and not a function, `flattened_table` might be a more fitting name.